From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Sun, 16 Feb 2025 01:13:04 +0100
Subject: [PATCH] Directly use the pre-filtered ticking chunks list as the
 output

This patch uses already pre filtered chunks, which completely skips the isChunkNearPlayer check

diff --git a/net/minecraft/server/level/ChunkMap.java b/net/minecraft/server/level/ChunkMap.java
index 0813505f96a1a6bb2cbfb05a97c90241f0c5b1c8..051d839fece06cac8ad79d8d936237a66cd01ae4 100644
--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -739,13 +739,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
         }
         // Leaf end - Use ensureCapacity to pre-populate the size of ticking chunks list output
         for (int i = 0; i < size; ++i) {
-            final LevelChunk levelChunk = raw[i];
-
-            if (!this.isChunkNearPlayer((ChunkMap)(Object)this, levelChunk.getPos(), levelChunk)) {
-                continue;
-            }
-
-            output.add(levelChunk);
+            output.add(raw[i]); // Leaf - Directly use the pre-filtered ticking chunks list as the output
         }
     }
 
